website/frontend/src/components/MetaFields/new-meta-parser.ts

```diff
@@ -0,0 +1,354 @@
+import isObject from 'lodash.isobject';
+import isString from 'lodash.isstring';
+import setWith from 'lodash.setwith';
+
+export type MetaTypes =
+  | 'Enum'
+  | 'Struct'
+  | 'Primitive'
+  | 'Option'
+  | 'Vec'
+  | 'Result'
+  | 'Tuple'
+  | 'Array'
+  | 'BTreeMap'
+  | 'BTreeSet'
+  | string;
+
+export type MetaItem = {
+  type: MetaTypes;
+  name: string;
+  value: MetaItemValue;
+};
+
+export type MetaItemValue = string | MetaItem | MetaItem[] | Record<string, MetaItem>;
+
+export type MetaField = {
+  type: MetaTypes;
+  name: string;
+  label: string;
+};
+
+export function isMetaField(value: unknown): value is MetaField {
+  return (
+    !(typeof value == 'boolean') &&
+    !isString(value) &&
+    isObject(value) &&
+    'type' in value &&
+    'name' in value &&
+    'label' in value
+  );
+}
+
+function isMetaItem(value: unknown): value is MetaItem {
+  return isObject(value) && 'type' in value && 'value' in value && 'name' in value;
+}
+
+function isPrimitive(value: MetaItemValue) {
+  return isMetaItem(value) && value.type === 'Primitive';
+}
+
+export type MetaFieldsItem = MetaField | MetaFieldset;
+export type MetaFieldsItemStruct = Record<string, MetaFieldsItem>;
+
+export type MetaFieldset = {
+  __name: string;
+  __path: string;
+  __type: string;
+  __select: boolean;
+  __fields: MetaFieldsItemStruct;
+};
+
+export function isMetaFieldset(value: unknown): value is MetaFieldset {
+  return (
+    !(typeof value == 'boolean') &&
+    !isString(value) &&
+    isObject(value) &&
+    '__name' in value &&
+    '__type' in value &&
+    '__select' in value &&
+    '__fields' in value
+  );
+}
+
+export type MetaFieldsValues = {
+  [key: string]: string | MetaFieldsValues;
+};
+
+export type MetaFieldsStruct = {
+  __root: MetaFieldset | null;
+  __values: MetaFieldsValues | null;
+};
+
+type StackItem = {
+  kind: MetaTypes;
+  path: string[];
+  value: MetaItemValue;
+  name: string;
+} | null;
+
+const NoneField = { None: { type: 'Primitive', name: 'None', value: 'None' } };
+
+function processFields(data: MetaItem, _path?: string[], isField = false): StackItem | null {
+  const path = _path || [];
+
+  if (data.type === 'Enum') {
```

Comment:
You can use switch - case.

Reply:
Agree, something like:
```suggestion
    switch (type) {
    case 'Enum':
    case 'Option':
    case 'Result':
      return {
        kind: data.type,
        path: [...path, data.name],
        value: data.value,
        name: data.name,
      }
    }
```